diff --git a/NNBitcoins/.flowconfig b/NNBitcoins/.flowconfig
index 7d608958..0b611f07 100644
--- a/NNBitcoins/.flowconfig
+++ b/NNBitcoins/.flowconfig
@@ -50,7 +50,5 @@ suppress_comment=\\(.\\|\n\\)*\\$FlowIssue\\((\\(<VERSION>\\)? *\\(site=[a-z,_]*
 suppress_comment=\\(.\\|\n\\)*\\$FlowFixedInNextDeploy
 suppress_comment=\\(.\\|\n\\)*\\$FlowExpectedError
 
-unsafe.enable_getters_and_setters=true
-
 [version]
-^0.61.0
+^0.63.0
diff --git a/NNBitcoins/package.json b/NNBitcoins/package.json
index e87dc332..704d7a54 100644
--- a/NNBitcoins/package.json
+++ b/NNBitcoins/package.json
@@ -7,6 +7,6 @@
   },
   "dependencies": {NNBitcoins
     "react": "16.2.0",
-    "react-native": "0.52.2"
+    "react-native": "0.53.0"
   }
 }
